### ISSUE TITLE
`prefer-global-this` should ignore `typeof window === 'undefined'`

### ISSUE THREAD

**Issue by belgattitude** [sindresorhus/eslint-plugin-unicorn#2468]
Hey thanks for the plugin, it's so useful !

I recently updated to v56 and was surprised about the autofix

 It's very common to have

```js
const isServer = typeof window === 'undefined';
const isBrowser = typeof window !== 'undefined'
```

It fixes to

```js
const isServer = typeof globalThis === 'undefined';
const isBrowser = typeof globalThis !== 'undefined';
```

Which isn't valid

Interested to know your advices.

PS: Thanks to @fregant comment below: `window' in globalThis`

**Comment by fregante** [comment]
Yeah this sounds like it should be an exception. 

For now you can exclude it or use `'window' in globalThis` and then `!isBrowser`

**Comment by belgattitude** [comment]
If there isn’t any drawbacks, would it be possible to consider your suggested code in the autofix ?

how could we proceed. For me this issue  looks a dealbreaker

**Comment by axetroy** [comment]
`typeof window` is a shorthand way of writing `typeof globalThis.window`

So it should not emit an error. This seems reasonable. same as `self`
